src/executor/mod.rs

```diff
@@ -0,0 +1,376 @@
+            .field("strategy", &self.strategy)
+            .finish()
+    }
+}
+
+#[derive(Debug)]
+pub struct NoStrategy;
+pub enum HasStrategy {
+    ExecuteDirectly,
+    #[cfg(feature = "tokio")]
+    SpawnBlocking(tokio::runtime::Handle),
+    Custom(CustomClosure),
+}
+
+impl std::fmt::Debug for HasStrategy {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::ExecuteDirectly => write!(f, "ExecuteDirectly"),
+            #[cfg(feature = "tokio")]
+            Self::SpawnBlocking(handle) => f.debug_tuple("SpawnBlocking").field(handle).finish(),
+            Self::Custom(_) => f.debug_tuple("Custom").finish(),
+        }
+    }
+}
+
+impl<Strategy> ExecutorBuilder<Strategy> {
+    /// Set the max number of simultaneous futures processed by this executor.
+    ///
+    /// If this number is exceeded, the executor will wait to execute the
+    /// input closure until a permit can be acquired.
+    ///
+    /// A good value tends to be the number of cpu cores on your machine.
+    ///
+    /// ## Default
+    /// No maximum concurrency.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # fn run() {
+    /// vacation::init()
+    ///         .max_concurrency(10)
+    ///         .execute_directly()
+    ///         .install()
+    ///         .unwrap();
+    /// # }
+    #[must_use = "doesn't do anything unless used with a strategy"]
+    pub fn max_concurrency(self, max_task_concurrency: usize) -> Self {
+        Self {
+            max_concurrency: Some(max_task_concurrency),
+            ..self
+        }
+    }
+    /// Initializes a new (non-op) global strategy to wait in the current context.
+    ///
+    /// This is effectively a non-op wrapper that adds no special handling for the sync future
+    /// besides optional concurrency control.
+    ///
+    /// This is the default strategy if nothing is initialized, with no max concurrency.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # async fn run() {
+    /// vacation::init().execute_directly().install().unwrap();
+    /// # }
+    /// ```
+    #[must_use = "doesn't do anything unless install()-ed"]
+    pub fn execute_directly(self) -> ExecutorBuilder<HasStrategy> {
+        ExecutorBuilder::<HasStrategy> {
+            strategy: HasStrategy::ExecuteDirectly,
+            max_concurrency: self.max_concurrency,
+        }
+    }
+
+    /// Initializes a new global strategy to execute input closures by blocking on them inside the
+    /// tokio blocking threadpool via Tokio's [`spawn_blocking`].
+    ///
+    /// Requires `tokio` feature.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # async fn run() {
+    /// // this will include no concurrency limit when explicitly initialized
+    /// // without a call to [`concurrency_limit()`]
+    /// vacation::init().spawn_blocking().install().unwrap();
+    /// # }
+    /// ```
+    /// [`spawn_blocking`]: tokio::task::spawn_blocking
+    ///
```

Comment:
should document that this will panic if called outside of a Tokio runtime (which actually might happen, if you're doing this before you "enter" the runtime in your own main method)